### PR TITLE
Improve links in development docs

### PR DESCRIPTION
**Pull request by surajnarwade**
* Link to >1 install doc, depending on situation
* Link to official site for knative-wide developer info

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Goals:
 4. Use consistent verbs, nouns, and flags for various commands
 5. Be easily extended via a plugin mechanism (similar to Kubectl) to allow for experimentations and customization
 
+
```
Comment (cppforlife) [inline README.md]: being picky here... no need for a newline.

File: README.md [hunk 2]
```diff
@@ -39,8 +40,8 @@ describe get
 
 # Developers
 
-If you'd like to contribute, please see
-[CONTRIBUTING](https://github.com/knative/docs/blob/master/contributing/CONTRIBUTING.md)
+* If you'd like to contribute, please see
```
Comment (cppforlife) [inline README.md]: links change ok, but lets not make them bullet points.

File: docs/DEVELOPMENT.md [hunk 3]
```diff
@@ -31,15 +31,14 @@ You must install these tools:
...
-   - You can also setup using limited install guides for eg: Minikube/Minishift.
+1. [Set up a Knative](https://www.knative.dev/docs/install/), You can also setup using limited install guides like [Minikube](https://knative.dev/docs/install/knative-with-minikube/) or [Minishift](https://knative.dev/docs/install/knative-with-minishift/).
 
 ### Checkout your fork
 
 To check out this repository:
 
 1. Create your own [fork of this repo](https://help.github.com/articles/fork-a-repo/)
-1. Clone it to your machine:
+2. Clone it to your machine:
```
Comment (cppforlife) [inline docs/DEVELOPMENT.md]: github markdown automatically numbers things. keep as `1.`

**Comment by sixolet** [comment]
What's the rationale for moving this file? My conception of `/docs` is for documentation of `kn` and the libraries in `pkg`, and `DEVELOPMENT.md` is more like "documentation of how to contribute".

Open to other ideas, but I'd rather hear the rationale.

**Comment by sixolet** [comment]
Also, required to merge: change the title of the PR to something descriptive enough someone scanning  commit first-line strings will be able to understand an overview of what and why the change is.

For example: "move DEVELOPMENT.md from root to docs for consistency with..." or whatnot.

**Comment by rhuss** [comment]
> For example: "move DEVELOPMENT.md from root to docs for consistency with..." or whatnot.

fwiw: I can recommend [commitizen](https://www.npmjs.com/package/commitizen) style of commit messages (i.e. like in "refactor(docs): Move DEVELOPMENT.md to docs/ directory"). Although I'm not really using tooling for this kind of commit style of "type(scope): short description" but it helps to stay concise, and, if everyone agrees on this style of commit messages, also helps in auto-generating rich changelog entries. Even when projects don't agree I still find it useful so I hope you don't mind when I personally stick to these conventions ;-)

"types" is a limited set to help categorization like

* feat
* fix
* docs
* refactor
* style
* test
* perf
* chore

This [blog](https://dev.bleacherreport.com/how-we-use-commitizen-to-clean-up-commit-messages-a16790dcd2fd) has also some addtional background information.

**Comment by surajnarwade** [comment]
sorry for delayed response, I will rebase and update my PR today

File: README.md (hunk 2 above)
Comment (surajnarwade) [reply README.md]: :+1:

File: docs/DEVELOPMENT.md (hunk 3 above)
Comment (surajnarwade) [reply docs/DEVELOPMENT.md]: :+1:  cool TIL

**Comment by surajnarwade** [comment]
@sixolet , not specific reason but it's good to have all docs in one directory, so if in future if you plan to have github pages, it becomes easy.

**Comment by surajnarwade** [comment]
Hey @rhuss @sixolet  @cppforlife 
 review needed :smile_cat:

**Comment by rhuss** [comment]
thanks for the typo fixes ! wrt/ moving to docs, I don't have strong feelings. We can leave it top-level (reserving docs for end user docs), or move it docs/ to collect all docs in one place.

**Comment by surajnarwade** [comment]
@rhuss , okay let me know what should I do ?

this thing came in my mind since we do same here https://openshiftdo.org/development/

**Comment by rhuss** [comment]
> @rhuss , okay let me know what should I do ?

I'd suggest keeping the DEVELOPMENT.md top-level as long as we don't have more than a single file as development documentation. If it tends to grow we can move it to docs or a sub-dir of docs. Is easier to spot when being top-level and doesn't pollute much if its only one file.

**Comment by surajnarwade** [comment]
@rhuss , moved file back to root dir.

**Comment by rhuss** [comment]
@surajnarwade Looks like you have to fix the link in the top-level README to the original location

```
0606 14:08:21.494] ----------------------------------------------
I0606 14:08:21.494] ---- Checking links in the markdown files ----
I0606 14:08:21.494] ----------------------------------------------
I0606 14:08:25.293] README.md
I0606 14:08:25.294] 	ERROR	docs/DEVELOPMENT.md
I0606 14:08:25.294] 		Stat docs/DEVELOPMENT.md: no such file or directory
I0606 14:08:25.300] ============================
I0606 14:08:25.301] ==== BUILD TESTS FAILED ====
I0606 14:08:25.301] ============================
```

**Comment by surajnarwade** [comment]
@sixolet , I have updated as your suggestions

**Pull request merged**